### PR TITLE
fix(booking): default a bare quantity-tracked scan to all remaining units on partial check-in/out

### PR DESCRIPTION
**Pull request by carlosvirreira**
## What & why

The companion app checks a quantity-tracked asset in/out by scanning it — the request carries only the `assetId`, with no per-unit count. Two bugs on the current live 1.1.0 app (#2701):

- **Check-in** (Scan / Select to Check In) throws `Quantity-tracked assets must include at least one non-zero disposition (returned, consumed, lost, or damaged)`.
- **Check-out** (Scan / Select to Check Out) silently checks out **1 unit** no matter how many are booked, leaving the rest stuck reserved.

Web never hits these because the check-in/out drawer collects an explicit per-unit disposition before submitting.

## Fix — server-only, no app release

A **bare** disposition — one that arrived via the legacy `assetIds[]` bucket rather than an explicit `checkins[]` / `checkouts[]` payload — for a `QUANTITY_TRACKED` asset now resolves to **all remaining units** for that asset, computed inside the transaction under the existing `lockAssetForQuantityUpdate` row lock:

- **`partialCheckoutBooking`** — resolves the claim to the per-asset remaining and records it in `PartialBookingCheckout.quantities[]`.
- **`partialCheckinBooking`** — exempts bare scans from the non-zero-disposition guard and defaults to **return-all** (`TWO_WAY`) / **consume-all** (`ONE_WAY`) from the asset's `consumptionType`.

This mirrors what the **Check Out All** / **Check In All** buttons already do. **Explicit per-unit payloads are untouched** — a genuine "2 of 5" still records 2.

The check-out **delegate gate is deliberately left alone**: a bare scan keeps its sentinel `quantity: 1`, so a booked-more-than-1 asset never satisfies `qtyClaimsCoverFullRemaining` and never delegates to the full `checkoutBooking` with a wrong count. That avoids a `PartialBookingCheckout` "split-brain" where a fully-checked-out asset would be recorded as 1 unit and corrupt the later check-in remaining math.

Because the fix is server-side, it reaches every companion 1.1.0 install on deploy — no App Store release. The native quantity / disposition picker (so operators can split returned/consumed/lost/damaged and take partial counts) is the planned follow-up UI.

## Tests

- `partialCheckoutBooking`: bare booked-50 scan → all 50 out, `quantities: [50]`, and no delegation; explicit `quantity: 1` still records exactly 1.
- `partialCheckinBooking`: bare scan → one RETURN log for all remaining (`TWO_WAY`); one CONSUME log for all remaining (`ONE_WAY`).
- Full booking test suite green (733 passed); typecheck, ESLint and Prettier clean.

Refs #2701

## Summary by CodeRabbit

* **Bug Fixes**
  * Quantity-tracked bookings now correctly interpret scan-only (assetIds-only) dispositions as consuming/returning **all remaining** instead of a sentinel unit.
  * Explicit per-quantity values provided by the request are still applied exactly as entered.
  * Prevents re-scans that would result in zero-quantity consumption/return records.
* **Tests**
  * Added/expanded unit coverage for quantity-tracked “bare scan” checkout and check-in flows, including correct consume/return behavior for different asset types and remaining-quantity edge cases.

### CODE REVIEW

File: apps/webapp/app/modules/booking/service.server.ts [hunk 1]
```diff
@@ -5066,6 +5090,26 @@ export async function partialCheckinBooking({
...
+        // default to "all remaining" — return all for a returnable (TWO_WAY)
+        // asset, consume all for a consumable (ONE_WAY) one. Mirrors
+        // checkinBooking's big-button default. Only bare ids reach here with a
+        // zero disposition (the guard above rejects an explicit zero), so this
+        // never overrides an operator's explicit returned/consumed split.
+        if (
+          bareCheckinAssetIds.has(disp.assetId) &&
+          sumDisposition(disp) === 0 &&
+          cap > 0
+        ) {
```
[the remark this answers is not in the document]
Comment (carlosvirreira) [reply apps/webapp/app/modules/booking/service.server.ts:5102]: Good catch — fixed in 2ffe07111. A bare check-in scan whose asset has `cap === 0` (a re-scan of an already fully-reconciled asset) now throws `Cannot check in "<title>" — no units remain to check in on this booking.` inside the tx, instead of writing a no-op `PartialBookingCheckin` + event. This matches the rejection an explicit re-scan already gets via `claimed > cap`. Added a bare-re-scan regression test.

File: apps/webapp/app/modules/booking/service.server.ts [hunk 2]
```diff
@@ -6277,6 +6327,15 @@ export async function partialCheckoutBooking({
...
         }
 
+        // Bare QUANTITY_TRACKED scan (assetIds-only, no explicit count):
+        // resolve to "all remaining" for this asset — the per-asset `cap`
+        // computed under the row lock above. Mutating the disposition so the
+        // `PartialBookingCheckout.quantities[]` ledger write below records the
+        // real units, not the sentinel 1. Mirrors the "Check Out All" default;
+        // explicit per-unit `quantity` payloads are left untouched.
+        if (disp.defaultAllRemaining) {
+          disp.quantity = cap;
```
[the remark this answers is not in the document]
Comment (carlosvirreira) [reply apps/webapp/app/modules/booking/service.server.ts:6354]: Fixed in 2ffe07111. The sentinel is now only replaced with `cap` when `cap > 0`; on a re-scan of a fully-checked-out asset (`cap === 0`) the disposition keeps `quantity: 1`, so the existing `claimed > cap` guard throws `Only 0 units left…` — no more `quantities: [0]` row or audit events. Added a bare-re-scan regression test.

File: apps/webapp/app/modules/booking/service.server.test.ts [hunk 3]
```diff
@@ -6019,6 +6019,129 @@ describe("partialCheckinBooking — qty-tracked dispositions", () => {
...
+    expect(consumptionLogService.createConsumptionLog).toHaveBeenCalledWith(
+      expect.objectContaining({
+        assetId: mockQtyAssetId,
+        category: "CONSUME",
+        quantity: 10,
+        bookingId: mockQtyBookingId,
+      })
+    );
+  });
+
```
[the remark this answers is not in the document]
Comment (carlosvirreira) [reply apps/webapp/app/modules/booking/service.server.test.ts:6144]: Added in 95727c629. The two existing bare-check-in tests intentionally stay on the partial path (that is the new code under test); this adds a single-QT-asset full-coverage test that hits the early-exit → delegate. It asserts the batch is accepted and completes (`isComplete: true`), confirming a bare id now reaches the delegate instead of the non-zero-disposition guard it used to hit. I scoped it to that assertion rather than re-asserting `checkinBooking`'s internal `ConsumptionLog` — that all-remaining default is covered by `checkinBooking`'s own tests and would just re-test that function under `partialCheckinBooking`'s mocks.

**Review by DonKoko — APPROVED** [review]
